_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -0,0 +1,1191 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "mfx_common.h"
+
+#if defined(MFX_ENABLE_AV1_VIDEO_DECODE)
+
+#include "mfx_session.h"
+#include "mfx_av1_dec_decode.h"
+
+#include "mfx_task.h"
+
+#include "mfx_common_int.h"
+#include "mfx_common_decode_int.h"
+#include "mfx_vpx_dec_common.h"
+
+#include "mfx_umc_alloc_wrapper.h"
+
+#include "umc_av1_dec_defs.h"
+#include "umc_av1_frame.h"
+#include "umc_av1_utils.h"
+
+#include "libmfx_core_hw.h"
+
+#include "umc_av1_decoder_va.h"
+
+#include <algorithm>
+
+#include "vm_sys_info.h"
+
+#if defined(MFX_VA)
+#include "umc_h265_va_supplier.h"
+#if defined(MFX_ENABLE_CPLIB) || !defined(MFX_PROTECTED_FEATURE_DISABLE)
+#include "umc_va_linux_protected.h"
+#endif
+#endif
+
+inline
+mfxU32 CalculateAsyncDepth(eMFXPlatform platform, mfxVideoParam *par)
+{
+    mfxU32 asyncDepth = par ? par->AsyncDepth : 0;
+    if (!asyncDepth)
+    {
+        asyncDepth = (platform == MFX_PLATFORM_SOFTWARE) ? vm_sys_info_get_cpu_num() : MFX_AUTO_ASYNC_DEPTH_VALUE;
+    }
+
+    return asyncDepth;
+}
+
+namespace MFX_VPX_Utility
+{
+    inline
+    mfxU16 MatchProfile(mfxU32)
+    {
+        return MFX_PROFILE_AV1_MAIN;
+    }
+
+#if defined (MFX_VA_WIN)
+    const GUID DXVA_Intel_ModeAV1_VLD =
+    { 0xca44afc5, 0xe1d0, 0x42e6, { 0x91, 0x54, 0xb1, 0x27, 0x18, 0x6d, 0x4d, 0x40 } };
+
+    const GUID DXVA_Intel_ModeAV1_VLD_420_10b =
+    { 0xf9a16190, 0x3fb4, 0x4dc5, { 0x98, 0x46, 0xc8, 0x75, 0x1f, 0x83, 0xd6, 0xd7 } };
+#endif
+
+    inline
+    bool CheckGUID(VideoCORE* core, eMFXHWType type, mfxVideoParam const* par)
+    {
+        mfxVideoParam vp = *par;
+        mfxU16 profile = vp.mfx.CodecProfile & 0xFF;
+        if (profile == MFX_PROFILE_UNKNOWN)
+        {
+            profile = MatchProfile(vp.mfx.FrameInfo.FourCC);;
+            vp.mfx.CodecProfile = profile;
+        }
+
+#if defined (MFX_VA_LINUX)
+        if (core->IsGuidSupported(DXVA_Intel_ModeAV1_VLD, &vp) != MFX_ERR_NONE)
+            return false;
+
+        switch (profile)
+        {
+            case MFX_PROFILE_AV1_MAIN:
+                return true;
+            default:
+                return false;
+        }
+#else
+        core; type; par;
```

Comment:
`core; type; par;`

This actually does not work. You need `(void)core;`.

Reply:
I have changed and uploaded it. Thanks.